packages/block-library/src/gallery/block.json

```diff
@@ -50,6 +50,10 @@
 		"linkTo": {
 			"type": "string",
 			"default": "none"
+		},
+		"attachmentCaptions": {
+			"type": "array",
```

Comment:
I wonder if it'd be better to structure it as an object that maps from the imageId to the attachmentCaption. That way there wouldn't be concerns about keeping the array in order if a new image is inserted into the gallery.

Reply:
it is an array of objects:

```
attachmentCaptions: newImages.map( ( newImage ) => ( {
	id: newImage.id,
	caption: newImage.caption,
} ) ),
```

we already don't care about the image order.